src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -313,62 +313,75 @@ def k8s_install_cli(cmd, client_version='latest', install_location=None):
-    # Ensure that the SPN exists
-    principal_obj = _ensure_aks_service_principal(cmd.cli_ctx, service_principal, client_secret, subscription_id,
-                                                  dns_name_prefix, location, connector_name)
-    client_secret = principal_obj.get('client_secret')
-    service_principal = principal_obj.get('service_principal')
     # Get the TenantID
     profile = Profile(cli_ctx=cmd.cli_ctx)
     _, _, tenant_id = profile.get_login_credentials()
     # Check if we want the linux connector
     if os_type.lower() in ['linux', 'both']:
-        _helm_install_aci_connector(url_chart, connector_name, service_principal, client_secret,
+        _helm_install_aci_connector(image_tag, url_chart, connector_name, service_principal, client_secret,
                                     subscription_id, tenant_id, rgkaci.name, location,
                                     node_prefix + '-linux', 'Linux')
 
     # Check if we want the windows connector
     if os_type.lower() in ['windows', 'both']:
-        _helm_install_aci_connector(url_chart, connector_name, service_principal, client_secret,
+        _helm_install_aci_connector(image_tag, url_chart, connector_name, service_principal, client_secret,
                                     subscription_id, tenant_id, rgkaci.name, location,
                                     node_prefix + '-win', 'Windows')
 
 
-def _helm_install_aci_connector(url_chart, connector_name, service_principal, client_secret,
-                                subscription_id, tenant_id, aci_resource_group, aci_region,
-                                node_name, os_type):
-    image_tag = 'latest'
+def _helm_install_aci_connector(image_tag, url_chart, connector_name, service_principal,
+                                client_secret, subscription_id, tenant_id, aci_resource_group,
+                                aci_region, node_name, os_type):
     node_taint = 'azure.com/aci'
     helm_release_name = connector_name.lower() + "-" + os_type.lower()
     logger.warning("Deploying the ACI connector for '%s' using Helm", os_type)
     try:
-        subprocess.call(["helm", "install", url_chart, "--name", helm_release_name, "--set", "env.azureClientId=" +
-                         service_principal + ",env.azureClientKey=" + client_secret + ",env.azureSubscriptionId=" +
-                         subscription_id + ",env.azureTenantId=" + tenant_id + ",env.aciResourceGroup=" +
-                         aci_resource_group + ",env.aciRegion=" + aci_region + ",image.tag=" + image_tag +
-                         ",env.nodeName=" + node_name + ",env.nodeTaint=" + node_taint + ",env.nodeOsType=" + os_type])
+        values = "image.tag={},env.nodeName={},env.nodeTaint={},env.nodeOsType={}".format(image_tag, node_name, node_taint, os_type)  # pylint: disable=line-too-long
+
+        if service_principal is not None:
```

Comment:
Would this be better as just 

```python
if not service_principal:
```

That's the pythonic way, unless we specifically want identity comparison with `None` here.

Reply:
do you mean

`if not service_principal:`

Reply:
D'oh yes of course that's what I meant.